### PR TITLE
Added authentication instruction to public ECR

### PR DESCRIPTION
**Pull request by bhautikpip**
*Issue #, if available:*

*Description of changes:*

Modified README to add authentication instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,13 +82,19 @@ GOOS=linux GOARCH=arm64 go build -ldflags "-s -w" -o xray cmd/tracing/daemon.go
 As of Aug 31, 2020, windows and darwin builds for ARM64 are not supported by `go build`.
 
 ## Pulling X-Ray Daemon image from ECR Public Gallery
+Before pulling an image authenticate your docker client to the Amazon ECR public registry. For registry authentication options follow this [link](https://docs.aws.amazon.com/AmazonECR/latest/public/public-registries.html#public-registry-auth)
+
+Run below command to authenticate to public ECR registry using `get-login-password` (AWS CLI)
+``
+aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+``
```
Comment (willarmiros) [inline README.md:90]:
```suggestion
Run below command to authenticate to public ECR registry using `get-login-password` (AWS CLI)

``
aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
``
```

Comment (willarmiros) [reply README.md:90]: Adds newline so command will be rendered on separate line.

**Review by willarmiros — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -82,13 +82,20 @@ GOOS=linux GOARCH=arm64 go build -ldflags "-s -w" -o xray cmd/tracing/daemon.go
 As of Aug 31, 2020, windows and darwin builds for ARM64 are not supported by `go build`.
 
 ## Pulling X-Ray Daemon image from ECR Public Gallery
+Before pulling an image authenticate your docker client to the Amazon ECR public registry. For registry authentication options follow this [link](https://docs.aws.amazon.com/AmazonECR/latest/public/public-registries.html#public-registry-auth)
```
Comment (anuraaga) [inline README.md:85]: I think "you should authenticate" - I guess it still works without, just with low quota.

**Review by anuraaga — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (bhautikpip) [reply README.md:85]: Not sure if pulling without an authentication works as per ECR doc.(https://docs.aws.amazon.com/AmazonECR/latest/public/docker-pull-ecr-image.html). As per my understanding you have to authenticate docker client with public ECR in order to use `docker pull` command

Comment (anuraaga) [reply README.md:85]: Hmm the docs aren't so clear but the blog posts have all mentioned anonymous pull with a smaller quota

https://aws.amazon.com/blogs/aws/amazon-ecr-public-a-new-public-container-registry/

```
Anyone who pulls images anonymously will get 500 GB of free data bandwidth each month, after which they can sign up or sign in to an AWS account to get more. Simply authenticating with an AWS account increases free data bandwidth up to 5 TB each month when pulling images from the internet.
```

Comment (bhautikpip) [reply README.md:85]: I see. Nice!

**Pull request merged**